nsxt/resource_nsxt_policy_transport_zone.go

```diff
@@ -0,0 +1,291 @@
+	if err != nil {
+		return err
+	}
+	if exists {
+		return fmt.Errorf("resource with ID %s already exists", id)
+	}
+
+	// Create the resource using PATCH
+	log.Printf("[INFO] Creating TransportZone with ID %s under site %s enforcement point %s", id, siteID, epID)
+	err = policyTransportZonePatch(siteID, epID, id, d, m)
+	if err != nil {
+		return handleCreateError("TransportZone", id, err)
+	}
+
+	d.SetId(id)
+	d.Set("nsx_id", id)
+
+	return resourceNsxtPolicyTransportZoneRead(d, m)
+}
+
+func resourceNsxtPolicyTransportZoneRead(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+	tzClient := enforcement_points.NewTransportZonesClient(connector)
+
+	id, siteID, epID, err := policyTransportZoneIDTuple(d, m)
+	if err != nil {
+		return err
+	}
+
+	obj, err := tzClient.Get(siteID, epID, id)
+	if err != nil {
+		return handleReadError(d, "TransportZone", id, err)
+	}
+
+	d.Set("site_id", siteID)
+	d.Set("enforcement_point_id", epID)
+	d.Set("display_name", obj.DisplayName)
+	d.Set("description", obj.Description)
+	setPolicyTagsInSchema(d, obj.Tags)
+	d.Set("nsx_id", id)
+	d.Set("path", obj.Path)
+	d.Set("revision", obj.Revision)
+	d.Set("is_default", obj.IsDefault)
+	d.Set("transport_type", obj.TzType)
+	d.Set("uplink_teaming_policy_names", obj.UplinkTeamingPolicyNames)
+
+	return nil
+}
+
+func resourceNsxtPolicyTransportZoneUpdate(d *schema.ResourceData, m interface{}) error {
+	id, siteID, epID, err := policyTransportZoneIDTuple(d, m)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Updateing TransportZone with ID %s", id)
+	err = policyTransportZonePatch(siteID, epID, id, d, m)
+	if err != nil {
+		return handleUpdateError("TransportZone", id, err)
+	}
+
+	return resourceNsxtPolicyTransportZoneRead(d, m)
+}
+
+func resourceNsxtPolicyTransportZoneDelete(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+	tzClient := enforcement_points.NewTransportZonesClient(connector)
+
+	id, siteID, epID, err := policyTransportZoneIDTuple(d, m)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Deleting TransportZone with ID %s", id)
+	err = tzClient.Delete(siteID, epID, id)
+	if err != nil {
+		return handleDeleteError("TransportZone", id, err)
+	}
+
+	return nil
+}
+
+func resourceNsxtPolicyTransportZoneImporter(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	importID := d.Id()
+	s := strings.Split(importID, "/")
+	rd, err := nsxtPolicyPathResourceImporterHelper(d, m)
+	if err == nil {
+		epID, err := getParameterFromPolicyPath("/enforcement-points/", "/transport-zones/", importID)
+		if err != nil {
+			return nil, err
+		}
+		d.Set("enforcement_point_id", epID)
+		siteID, err := getParameterFromPolicyPath("/sites/", "/enforcement-points/", importID)
+		if err != nil {
+			return nil, err
+		}
+		d.Set("site_id", siteID)
+		return rd, nil
+	} else if !errors.Is(err, ErrNotAPolicyPath) {
+		return rd, err
+	}
+
+	var siteID, epID, id string
```

Comment:
I believe for all new nested resources we should only support import by policy path. @ksamoray what do you think?

Reply:
Makes sense. It won't take any functionality from the user, and will keep things tidy on our side.

Reply:
Removed the all other import forms than via policy ID